app/uk/gov/hmrc/fileupload/write/envelope/Envelope.scala

```diff
@@ -213,7 +234,36 @@ sealed trait State {
         Xor.left(FileAlreadyProcessed)
       }).getOrElse(fileNotFoundError)
 
-  def checkCanStoreFile(fileId: FileId, fileRefId: FileRefId, files: Map[FileId, File]): CanResult =
+  def checkCanFileQuarantined(fileId: FileId, fileRefId: FileRefId, fileLength: Long, files: Map[FileId, File], envelope: Envelope): CanResult = {
+
+    def sizeToByte(size: String): Long = {
+      val sizeRegex = "([1-9][0-9]{0,3})([KB,MB]{2})".r
```

Comment:
This method should be **unit tested** on its own. Also 0 seems to be some terminating condition but it's not clear why it would be the case. If user provides incorrect value is it going to prevent uploading anything? As allowed size seems to be up to 0 in such a case? This is wrong. We should reject a request to create an envelope instead.